doc/pset.mediawiki

```diff
@@ -0,0 +1,429 @@
+| <tt>PSBT_ELEMENTS_GLOBAL_SCALAR = 0x00</tt>
+| <tt><32 byte scalar></tt>
+| A 32 byte Scalar Offset computed by a blinder.
+| None
+| No value. The value must have a length of zero.
+|
+| 0
+| 2
+|-
+| Elements Transaction Modifiable Flag
+| <tt>PSBT_ELEMENTS_GLOBAL_TX_MODIFIABLE = 0x01</tt>
+| None
+| No key data
+| <tt><8-bit uint></tt>
+| An 8 bit little endian unsigned integer as a bitfield for various elements specific transaction modification flags. Bit 0 is the PSBT Blinded flag and it is set to 1 to indicate that the PSET has not been blinded yet. Once all confidential values, rangeproofs, and asset surjection proofs have been attached to the PSET, it must be set to 0.
+|
+| 0
+| 2
+|}
+
+The currently defined elements per-input proprietary types are as folows:
+
+{|
+! Name
+! <tt><keytype></tt>
+! <tt><keydata></tt>
+! <tt><keydata></tt> Description
+! <tt><valuedata></tt>
+! <tt><valuedata></tt> Description
+! Versions Requiring Inclusion
+! Versions Requiring Exclusion
+! Versions Allowing Inclusion
+|-
+| Issuance Value
+| <tt>PSBT_ELEMENTS_IN_ISSUANCE_VALUE = 0x00</tt>
+| None
+| No key data
+| <tt><64-bit int></tt>
+| The explicit little endian 64-bit integer for the value of this issuance. This is mutually exclusive with <tt>PSBT_ELEMENTS_IN_ISSUANCE_VALUE_COMMITMENT</tt>
+|
+| 0
+| 2
+|-
+| Issuance Value Commitment
+| <tt>PSBT_ELEMENTS_IN_ISSUANCE_VALUE_COMMITMENT = 0x01</tt>
+| None
+| No key data
+| <tt><33 byte commitment></tt>
+| The 33 byte Value Commitment. This is mutually exclusive with <tt>PSBT_IN_ISSUANCE_VALUE</tt>.
+|
+| 0
+| 2
+|-
+| Issuance Value Rangeproof
+| <tt>PSBT_ELEMENTS_IN_ISSUANCE_VALUE_RANGEPROOF = 0x02</tt>
+| None
+| No key data
+| <tt><rangeproof></tt>
+| The rangeproof
+|
+| 0
+| 2
+|-
+| Issuance Inflation Keys Rangeproof
+| <tt>PSBT_ELEMENTS_IN_ISSUANCE_KEYS_RANGEPROOF = 0x03</tt>
+| None
+| No key data
+| <tt><rangeproof></tt>
+| The rangeproof
+|
+| 0
+| 2
+|-
+| Peg-in Transaction
+| <tt>PSBT_ELEMENTS_IN_PEG_IN_TX = 0x04</tt>
+| None
+| No key data
+| <tt><tx></tt>
+| The Peg-in Transaction serialized without witnesses.
+|
+| 0
+| 2
+|-
+| Peg-in Transaction Output Proof
+| <tt>PSBT_ELEMENTS_IN_PEG_IN_TXOUT_PROOF = 0x05</tt>
+| None
+| No key data
+| <tt><txoutproof></tt>
+| The transaction output proof for the Peg-in Transaction.
+|
+| 0
+| 2
+|-
+| Peg-in Genesis Hash
+| <tt>PSBT_ELEMENTS_IN_PEG_IN_GENESIS = 0x06</tt>
+| None
+| No key data
+| <tt><hash></tt>
+| The 32 byte genesis hash for the Peg-in Transaction.
+|
+| 0
+| 2
+|-
+| Peg-in Claim Script
+| <tt>PSBT_ELEMENTS_IN_PEG_IN_CLAIM_SCRIPT = 0x07</tt>
+| None
+| No key data
+| <tt><script></tt>
+| The claim script for the Peg-in Transaction.
+|
+| 0
+| 2
+|-
+| Peg-in Value
+| <tt>PSBT_ELEMENTS_IN_PEG_IN_VALUE = 0x08</tt>
+| None
+| No key data
+| <tt><64-bit int></tt>
+| The little endian 64-bit value of the peg-in for the Peg-in Transaction.
+|
+| 0
+| 2
+|-
+| Peg-in Witness
+| <tt>PSBT_ELEMENTS_IN_PEG_IN_WITNESS = 0x09</tt>
```

Comment:
The name makes it seem that the type of the value should Witness, but it is shown as a script. How does this work for segwit transactions?

Reply:
Maybe this question does not make sense, and pegin transactions input only have witness in ScriptSig.

Reply:
It's a `CScriptWitness`: https://github.com/ElementsProject/elements/blob/master/src/pegins.cpp#L521